saga-demo/conditional-transaction-demo/README.md

```diff
@@ -0,0 +1,170 @@
+# Conditional Transaction Demo
+This demo simulates a partial backend of an e-commerce application including four services:
+* payment
+* membership
+* inventory
+* supplier
+
+**Note** Please go through the dependency-free-transaction-demo first, before proceeding with this demo. 
+
+## Background
+Many e-commerce sites have membership concept, which will level up when a customer makes a certain amount of purchase.
+Moreover, as goods are sold and the stock drops to a certain level, the e-commerce company needs to fetch more goods from
+its suppliers. Both scenarios introduce conditional transactions. 
+
+## Workflow
+```
+                  /---> if total purchase >= 1K, level up membership
+start ---> make payment ---> dispatch product from inventory ---> end
+                                              \---> if product stock < 10, replenish product from supplier 
+```
+The conditions are business logic and belong to business service. Therefore, the decision to level up membership is made
+in payment service and the decision to replenish products is made in inventory service. Saga is to be informed of service
+decision by adding an additional element `sagaChildren` in service response JSON to indicate which services are to be invoked
```

Comment:
What if the sagaChildren failed? 
sagaChildren could be a bunch of services,  what if one of these child saga service call is failed.

Reply:
it will trigger compensation as usual